website/blog/2023-09-13-reason-react-ppx.md

```diff
@@ -0,0 +1,55 @@
+---
+title: `reason-react-ppx` is now on opam!
+---
+
+Today, we're releasing ReasonReact 0.12! The most important changes are renaming the JSX transform PPX from `reactjs-jsx-ppx` to `reason-react-ppx`, along with many other improvements:
+
+- `reason-react-ppx` now emits the React 17 [JSX transform](https://legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html).
+- prop locations have been greatly improved, resulting in much better editor integration.
+- check the full changes below for more details around other fixes and small breaking changes.
+
+### Locations from domProps in action (Thanks @jchavarri)
+![download](https://github.com/reasonml/reason-react/assets/3763599/db505599-8fee-4889-80a3-d2056ece291c)
+
+### new JSX transform (Thanks @anmonteiro)
+<img width="1508" alt="Screenshot 2023-09-13 at 16 11 17" src="https://github.com/reasonml/reason-react/assets/3763599/b7fec116-131a-48cd-8be1-1496544131d1">
+
+## What's changing?
+
+Track `reacjs-jsx-ppx` in the same repository as `reason-react`, and rename it to `reason-react-ppx`, a more suitable name with a few improvements:
+- Use the [new JSX transform](https://legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html) from React 17
+- Fix code-generation locations for JSX and props
+- Fix to allow using `React.memoCustomCompareProps`
+- Fix Type of optional prop not checked correctly in JSX
+- The ppx will be synced with reason-react (and tested together)
+
+## Updating to latest
+
+ReasonReact works best with Melange. [Get started here](https://melange.re/v1.0.0/getting-started/).
+
+Next:
+
+- Update opam repository with `opam update`
+- Install the latest versions: `opam install reason-react reason-react-ppx`
```

Comment:
Should this be

```suggestion
- Upgrade to the latest versions: `opam upgrade reason-react reason-react-ppx`
```

? If this is a flow where the user already had the `reason-react` package, I'm not sure that running `opam install` will perform the upgrade.

In case they haven't installed the packages, `opam upgrade` asks the users to install them first:

```
$ opam upgrade reason-react reason-react-ppx
reason-react-ppx is not installed. Install it? [Y/n]
```

Reply:
Technically

opam install reason-react-ppx
opam upgrade reason-react

Reply:
Yeah, maybe the first command already upgrades `reason-react` because of the `{ = version }` constraint. I suggested single `upgrade` command because it takes care of all cases at once.